sled-agent/src/illumos/zone.rs

```diff
@@ -425,11 +430,13 @@ impl Zones {
         // Ensure that a static IPv6 address has been allocated
         // to the Global Zone. Without this, we don't have a way
         // to route to IP addresses that we want to create in
-        // the non-GZ.
+        // the non-GZ. Note that we use a `/64` prefix, as all addresses
+        // allocated for services on this sled itself are within the underlay
+        // prefix. Anything else must be routed through Sidecar.
```

Comment:
Yeah, the original rationale for this was "Hey, I can't route to IPv6 addresses between zones unless the GZ has one. Give it an address arbitrarily" - hence the name. The use of the `/16` was totally wrong, for the reasons you mentioned.

Thanks for this cleanup!